### PR TITLE
Add backward compatibility with the previous firmware version

### PR DESCRIPTION
**Pull request by CedrikNikita**

### CODE REVIEW

File: src/Ae.js [hunk 1]
```diff
@@ -84,30 +84,36 @@ export default class Ae {
     rawTxHex: string,
     networkId: string = "ae_mainnet"
   ): Promise<string> {
+    let isNewAppVersion = true;
+    const appConfig = await this.transport.send(CLA, GET_APP_CONFIGURATION, 0x00, 0x00);
+    if (appConfig[1] === 0 && appConfig[2] < 4) {
+      isNewAppVersion = false;
+    }
```
Comment (davidyuk) [inline src/Ae.js]: Proposing to use more high-level method to detect version and name variable more semantic:
```suggestion
    const [major, minor] = (await this.getAppConfiguration()).version.split(".").map(n => +n);
    const acceptsTransactionLength = !(major === 0 && minor < 4);
```

File: src/Ae.js [hunk 2]
```diff
@@ -84,30 +84,36 @@ export default class Ae {
...
     let offset = 0;
+    const indent = 4 + 1 + (isNewAppVersion ? 4 : 0);
```
Comment (davidyuk) [inline src/Ae.js]: `indent` sounds similar to `offset` that is used near, maybe name it like `headerLength` instead?

File: src/Ae.js [hunk 3]
```diff
@@ -84,30 +84,36 @@ export default class Ae {
...
           : chunkSize
       );
       if (offset === 0) {
         buffer.writeUInt32BE(accountIndex, 0);
-        buffer.writeUInt32BE(rawTx.length, 4);
-        buffer.writeUInt8(networkIdBuffer.length, 8);
-        networkIdBuffer.copy(buffer, 4 + 4 + 1, offset, networkIdBuffer.length);
+        if (isNewAppVersion) buffer.writeUInt32BE(rawTx.length, 4);
+        buffer.writeUInt8(networkIdBuffer.length, isNewAppVersion ? 8 : 4);
+        networkIdBuffer.copy(buffer, indent, offset, networkIdBuffer.length);
```
Comment (davidyuk) [inline src/Ae.js]: I think we cam simplify offset calculation by using return values of `write...` functions:
```js
let offset = 0;
offset += buffer.writeUInt32BE(accountIndex, offset);
if (isNewAppVersion) offset += buffer.writeUInt32BE(rawTx.length, offset);
offset += buffer.writeUInt8(networkIdBuffer.length, offset);
offset += networkIdBuffer.copy(buffer, offset, 0, networkIdBuffer.length);
rawTx.copy(buffer, offset, 0, 150 - offset);
```
https://nodejs.org/api/buffer.html#buffer_buf_writeuint32be_value_offset
https://nodejs.org/api/buffer.html#buffer_buf_copy_target_targetstart_sourcestart_sourceend

File: src/Ae.js (hunk 1 above)
Comment (CedrikNikita) [reply src/Ae.js]: We actually should drop the getAppConfiguration method out of decorateAppAPIMethods to do so.

File: src/Ae.js (hunk 3 above)
Comment (CedrikNikita) [reply src/Ae.js]: this is not quite correct, but we can do it by using return values thats true.

File: src/Ae.js [hunk 4]
```diff
@@ -150,6 +149,8 @@ export default class Ae {
       });
   }
 
+  _getAppConfiguration = this.getAppConfiguration;
```
Comment (davidyuk) [inline src/Ae.js:151]: so, the idea is to don't wrap it with `decorateAppAPIMethods`

File: src/Ae.js [hunk 5]
```diff
@@ -84,33 +84,32 @@ export default class Ae {
...
-          4 + 4 + 1 + networkIdBuffer.length,
-          offset,
-          offset + chunkSize
-        );
+        offset = buffer.writeUInt32BE(accountIndex, offset);
+        if (acceptsTransactionLength) offset = buffer.writeUInt32BE(rawTx.length, offset);
+        offset = buffer.writeUInt8(networkIdBuffer.length, offset);
+        offset += networkIdBuffer.copy(buffer, offset, 0, networkIdBuffer.length);
+        rawTx.copy(buffer, offset, 0, 150 - offset);
+        offset = 0;
```
Comment (davidyuk) [inline src/Ae.js]: Use the same variable for the offset in rawTx and the offset in buffer doesn't look obvious, proposing to name them differently

File: src/Ae.js [hunk 6]
```diff
@@ -84,33 +84,32 @@ export default class Ae {
     rawTxHex: string,
     networkId: string = "ae_mainnet"
   ): Promise<string> {
+    const [major, minor] = (await this._getAppConfiguration()).version.split(".").map(n => +n);
+    const acceptsTransactionLength = !(major === 0 && minor < 4);
     let offset = 0;
+    const headerLength  = 4 + 1 + (acceptsTransactionLength ? 4 : 0);
```
Comment (davidyuk) [inline src/Ae.js:90]: We can naturally calculate this length by writing the header (that we have to do in the first chunk). So, proposing to write header firstly and then to take it length.

**Review by davidyuk — APPROVED** [review]

**Pull request merged**